src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -594,5 +594,23 @@ internal static Location FirstOrNone(this ImmutableArray<Location> items)
         {
             return items.IsEmpty ? Location.None : items[0];
         }
+
+        internal static bool SequenceEqual<TElement, TArg>(ImmutableArray<TElement> array1, ImmutableArray<TElement> array2, TArg arg, Func<TElement, TElement, TArg, bool> predicate)
+        {
+            if (array1.IsDefault || array2.IsDefault || array1.Length != array2.Length)
```

Comment:
>array1.IsDefault || array2.IsDefault [](start = 16, length = 36)

Is it important to allow these cases? Aren't those cases invalid arguments? And if they are not invalid, shouldn't `array1.IsDefault && array2.IsDefault` return `true`?